### PR TITLE
makes collection fallible

### PR DESCRIPTION
**Pull request by MarinPostma**
makes `Collection` operations fallible

### CODE REVIEW

File: src/collection.rs [hunk 1]
```diff
@@ -124,15 +162,24 @@ impl<T> Counter<T> {
...
+    type Error = Infallible;
+
+    fn push(&mut self, _elem: T) -> Result<(), Self::Error> {
         self.count = self.count.saturating_add(1);
+        Ok(())
     }
 
-    fn extend_from_slice(&mut self, elems: &[T]) where T: Clone {
+    fn extend_from_slice(&mut self, elems: &[T]) -> Result<(), Self::Error>
+    where T: Clone {
```
Comment (Kerollmops) [inline src/collection.rs]:
```suggestion
    where T: Clone
    {
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
It seems really clean, and not that complex in fact, most of those changes are just functions signatures!

File: src/collection.rs (hunk 1 above)
Comment (MarinPostma) [reply src/collection.rs]: good catch! spent 20 minutes changing them all... ;)

Comment (MarinPostma) [reply src/collection.rs]: fixed

**Pull request merged**